{
  "Software_Repo_MappingId": null,
  "Software_RepoId": "f6b863cd-8f1c-4583-8bc5-ee59e86e5cf1",
  "Software_Name": "Python 3.12.5 (64-bit)",
  "CVEID": "CVE-2018-1000117",
  "Summary": "Python Software Foundation CPython version From 3.2 until 3.6.4 on Windows contains a Buffer Overflow vulnerability in os.symlink() function on Windows that can result in Arbitrary code execution, likely escalation of privilege. This attack appears to be exploitable via a python script that creates a symlink with an attacker controlled name or location. This vulnerability appears to have been fixed in 3.7.0 and 3.6.5.",
  "LastUpdate": 0,
  "CVEE": {
    "CVEEId": "CVE-2018-1000117",
    "CVSS_V2_baseScore": 7.2,
    "CVSS_V2_Data": "{\"version\":\"2.0\",\"baseScore\":7.2,\"accessVector\":\"LOCAL\",\"vectorString\":\"AV:L/AC:L/Au:N/C:C/I:C/A:C\",\"authentication\":\"NONE\",\"integrityImpact\":\"COMPLETE\",\"accessComplexity\":\"LOW\",\"availabilityImpact\":\"COMPLETE\",\"confidentialityImpact\":\"COMPLETE\",\"scope\":null,\"attackVector\":null,\"baseSeverity\":null,\"userInteraction\":null,\"attackComplexity\":null,\"privilegesRequired\":null}",
    "CVSS_V3_baseScore": 6.7,
    "CVSS_V3_Data": "{\"version\":\"3.1\",\"baseScore\":6.7,\"accessVector\":null,\"vectorString\":\"CVSS:3.1/AV:L/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H\",\"authentication\":null,\"integrityImpact\":\"HIGH\",\"accessComplexity\":null,\"availabilityImpact\":\"HIGH\",\"confidentialityImpact\":\"HIGH\",\"scope\":\"UNCHANGED\",\"attackVector\":\"LOCAL\",\"baseSeverity\":\"MEDIUM\",\"userInteraction\":\"NONE\",\"attackComplexity\":\"LOW\",\"privilegesRequired\":\"HIGH\"}",
    "PublishedDate": "2018-03-07T14:29:00.28",
    "ModifyDate": "2024-11-21T03:39:40.98",
    "VulnStatus": "Modified",
    "JsonData": "{\"id\": \"CVE-2018-1000117\", \"summary\": \"Python Software Foundation CPython version From 3.2 until 3.6.4 on Windows contains a Buffer Overflow vulnerability in os.symlink() function on Windows that can result in Arbitrary code execution, likely escalation of privilege. This attack appears to be exploitable via a python script that creates a symlink with an attacker controlled name or location. This vulnerability appears to have been fixed in 3.7.0 and 3.6.5.\", \"created_at\": \"2018-03-07T14:29:00Z\", \"updated_at\": \"2024-11-21T03:39:40Z\", \"cvss\": {\"v2\": 7.2, \"v3\": 6.7}, \"vendors\": {\"python\": [\"python\"], \"microsoft\": [\"windows\"]}, \"cwes\": [\"CWE-120\"], \"raw_nvd_data\": {\"id\": \"CVE-2018-1000117\", \"metrics\": {\"cvssMetricV2\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"cvssData\": {\"version\": \"2.0\", \"baseScore\": 7.2, \"accessVector\": \"LOCAL\", \"vectorString\": \"AV:L/AC:L/Au:N/C:C/I:C/A:C\", \"authentication\": \"NONE\", \"integrityImpact\": \"COMPLETE\", \"accessComplexity\": \"LOW\", \"availabilityImpact\": \"COMPLETE\", \"confidentialityImpact\": \"COMPLETE\"}, \"acInsufInfo\": false, \"impactScore\": 10.0, \"baseSeverity\": \"HIGH\", \"obtainAllPrivilege\": false, \"exploitabilityScore\": 3.9, \"obtainUserPrivilege\": false, \"obtainOtherPrivilege\": false, \"userInteractionRequired\": false}], \"cvssMetricV31\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"cvssData\": {\"scope\": \"UNCHANGED\", \"version\": \"3.1\", \"baseScore\": 6.7, \"attackVector\": \"LOCAL\", \"baseSeverity\": \"MEDIUM\", \"vectorString\": \"CVSS:3.1/AV:L/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H\", \"integrityImpact\": \"HIGH\", \"userInteraction\": \"NONE\", \"attackComplexity\": \"LOW\", \"availabilityImpact\": \"HIGH\", \"privilegesRequired\": \"HIGH\", \"confidentialityImpact\": \"HIGH\"}, \"impactScore\": 5.9, \"exploitabilityScore\": 0.8}]}, \"published\": \"2018-03-07T14:29:00.280\", \"references\": [{\"url\": \"https://bugs.python.org/issue33001\", \"tags\": [\"Issue Tracking\", \"Patch\", \"Third Party Advisory\"], \"source\": \"cve@mitre.org\"}, {\"url\": \"https://github.com/python/cpython/pull/5989\", \"tags\": [\"Issue Tracking\", \"Patch\", \"Vendor Advisory\"], \"source\": \"cve@mitre.org\"}, {\"url\": \"https://bugs.python.org/issue33001\", \"tags\": [\"Issue Tracking\", \"Patch\", \"Third Party Advisory\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://github.com/python/cpython/pull/5989\", \"tags\": [\"Issue Tracking\", \"Patch\", \"Vendor Advisory\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}], \"vulnStatus\": \"Modified\", \"weaknesses\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"description\": [{\"lang\": \"en\", \"value\": \"CWE-120\"}]}], \"descriptions\": [{\"lang\": \"en\", \"value\": \"Python Software Foundation CPython version From 3.2 until 3.6.4 on Windows contains a Buffer Overflow vulnerability in os.symlink() function on Windows that can result in Arbitrary code execution, likely escalation of privilege. This attack appears to be exploitable via a python script that creates a symlink with an attacker controlled name or location. This vulnerability appears to have been fixed in 3.7.0 and 3.6.5.\"}, {\"lang\": \"es\", \"value\": \"Python Software Foundation CPython, desde la versi\\u00f3n 3.2 hasta la 3.6.4 en Windows, contiene una vulnerabilidad de desbordamiento de b\\u00fafer en la funci\\u00f3n os.symlink() de Windows que puede resultar en la ejecuci\\u00f3n de c\\u00f3digo arbitrario y en un escalado de privilegios. El ataque parece ser explotable mediante un script de python que crea un symlink con un nombre o ubicaci\\u00f3n controlado por un atacante. La vulnerabilidad parece haber sido solucionada en las versiones 3.7.0 y 3.6.5.\"}], \"lastModified\": \"2024-11-21T03:39:40.980\", \"configurations\": [{\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"C7B9CFAC-B5D1-4A16-A311-42487E31655D\", \"versionEndExcluding\": \"3.4.9\", \"versionStartIncluding\": \"3.2.0\"}, {\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"D1AAA7FB-F42F-480D-9549-EF8543876AFA\", \"versionEndExcluding\": \"3.5.6\", \"versionStartIncluding\": \"3.5.0\"}, {\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"A9862179-5DDB-4B35-A010-D6DEDCACAA5F\", \"versionEndExcluding\": \"3.6.5\", \"versionStartIncluding\": \"3.6.0\"}, {\"criteria\": \"cpe:2.3:a:python:python:3.7.0:beta1:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"6A7E7890-36B8-4259-AC7B-19FEADC23ADD\"}, {\"criteria\": \"cpe:2.3:a:python:python:3.7.0:beta2:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"573E7CAE-9884-44F1-8A3A-54081DB63618\"}, {\"criteria\": \"cpe:2.3:a:python:python:3.7.0:beta3:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"819DE67C-A7A6-46DE-A41F-7B7F805944A3\"}, {\"criteria\": \"cpe:2.3:a:python:python:3.7.0:beta4:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"BE8DDDF3-2695-493C-A21F-691ECD91EE79\"}, {\"criteria\": \"cpe:2.3:a:python:python:3.7.0:beta5:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"C45769F8-486F-4041-B059-7B0DD16F2E74\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:microsoft:windows:*:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"2CF61F35-5905-4BA9-AD7E-7DB261D2F256\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}], \"sourceIdentifier\": \"cve@mitre.org\"}}\n",
    "LastUpdate": 0,
    "VendorL": [
      "python",
      "microsoft"
    ],
    "ProductL": [
      "python",
      "windows"
    ],
    "Software_Repo_Mappings": []
  },
  "Software_Repo_Mapping_Conditions": [
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.4.9",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.2.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.5.6",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.5.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.6.5",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.6.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:3.7.0:beta1:*:*:*:*:*:*",
      "VersionEndExcluding": null,
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": null,
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:3.7.0:beta2:*:*:*:*:*:*",
      "VersionEndExcluding": null,
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": null,
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:3.7.0:beta3:*:*:*:*:*:*",
      "VersionEndExcluding": null,
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": null,
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:3.7.0:beta4:*:*:*:*:*:*",
      "VersionEndExcluding": null,
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": null,
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:3.7.0:beta5:*:*:*:*:*:*",
      "VersionEndExcluding": null,
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": null,
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    }
  ]
}